### PR TITLE
ref: Rename types and draft a changelog

### PR DESCRIPTION
**Pull request by Swatinem**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## 4.0.0
+
+This is a complete rewrite of the crate. It focuses on the `Mapping` type, which
+provides high level metadata about a proguard mapping file, and allows iterating
+over the contained `Record`s, and the `Mapper`, which allows re-mapping class
+names and entire frames, with support for inlined frames as well.
```
Comment (jan-auer) [inline CHANGELOG.md:8]: How would you feel about giving a few examples on how to migrate (e.g. which type to replace with which)?

**Comment by Swatinem** [comment]
hm, question is, would you rather `use proguard::ProgruardMapping` or use it fully qualified, like `proguard::Mapping`? I’m not really sure either way.

File: src/mapper.rs [hunk 2]
```diff
@@ -173,10 +173,10 @@ impl<'s> ProguardMapper<'s> {
     /// # Examples
     ///
     /// ```
-    /// use proguard::ProguardMapper;
+    /// use proguard::Mapper;
     ///
     /// let mapping = r#"android.arch.core.executor.ArchTaskExecutor -> a.a.a.a.c:"#;
-    /// let mapper = ProguardMapper::from(mapping);
+    /// let mapper = Mapper::from(mapping);
```
Comment (flub) [inline src/mapper.rs:179]: Since you asked for bikeshedding; this is probably nicer as `let mapper = proguard::Mapper` without the `use`.

**Comment by jan-auer** [comment]
I would 100% choose `use proguard::ProguardMapping`.

**Pull request merged**